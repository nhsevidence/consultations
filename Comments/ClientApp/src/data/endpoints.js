export const BaseUrl  = "/consultations";

export const Endpoints = {
	consultation: "/api/Consultation", // details for a single consultation
	consultations: "/api/Consultations", // list of consultations
	consultationList: "/api/ConsultationList", // list of consultations for the download page
	draftconsultation: "/api/DraftConsultation", // details for a single consultation that has never been published
	documents: "/api/Documents", // documents contained within consultation
	previewdraftdocuments: "/api/PreviewDraftDocuments", // documents contained within consultation
	previewpublisheddocuments: "/api/PreviewpublishedDocuments", // documents contained within consultation
	chapter: "/api/Chapter", // chapter content for a document
	previewchapter: "/api/PreviewChapter", // chapter content for a document in preview mode
	comments: "/api/Comments", // list of comments for a given URI, restricted by current user
	commentsreview: "/api/CommentsForReview", // list of comments for a given URI, restricted by current user
	editcomment: "/api/Comment/{0}", // edits an existing comment. id must be positive int, restricted by current user
	newcomment: "/api/Comment", // creates a new comment. body of message contains comment in json, restricted by current user
	editanswer: "/api/Answer/{0}", // edits an existing answer. id must be positive int, restricted by current user
	newanswer: "/api/Answer", // creates a new answer. body of message contains answer in json, restricted by current user
	review: "/api/Review/{0}", //list of all comments for a consultation (inc Consultation, Document, Chapter and Section levels) given a URI, restricted by current user
	user: "/api/User", // get details of currently logged in user
	submit: "/api/Submit", // submit on the review page.
	submitToOrgLead: "/api/SubmitToOrgLead", // submit to lead on the review page
	exportExternal: "/api/ExportExternal/{0}", //Creates an excel spread sheet for external users of the system that contains only their responses
	logging: "/api/Logging", // sends a message to the server for logging. expects the body to be the message and the loglevel to be in the querystring eg ?logLevel=Error
	questions: "/api/Questions", // get complete page structure for the admin of questions for a given consultation. /api/Questions?consultationId=22
	question: "/api/Question/{0}", // GET PUT for individual questions
	newquestion: "/api/Question", // POST for new questions
	organisation: "/api/Organisation", //GET checks for valid collation code and consultation id in querystring. returns OrganisationCode object
	organisationsession: "/api/Organisation/CreateOrganisationUserSession", //POST. checks for valid collation code and organisation authorisation id in querystring. returns a guid if successful. error otherwise.
	checkorganisationusersession: "/api/Organisation/CheckOrganisationUserSession", //GET. checks for valid session id for a given consultation id. returns a boolean for valid or not.
};
