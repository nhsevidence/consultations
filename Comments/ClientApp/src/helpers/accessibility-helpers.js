/**
 * Pull focus to an element, principally for accessibility
 * @param {string} selection - The ***ID ONLY*** If it's not an interactive element, the target will need an attribute of "tabIndex={-1}"
 */
// todo: needs test
export const pullFocusByQuerySelector = selection => {
	const el = document.querySelector(selection);
	if (el.getAttribute("tabindex") === null) {
		el.setAttribute("tabindex", "-1");
	}
	el.focus();
	console.log(`Pulling focus by querySelector to ${selection}`);
};

export const pullFocusById = selection => {
	selection = selection.replace("#", ""); // make sure there's no # being passed into here
	const el = document.getElementById(selection);
	if (el.getAttribute("tabindex") === null) {
		el.setAttribute("tabindex", "-1");
	}
	el.focus();
	console.log(`Pulling focus by ID to ${selection}`);
};
