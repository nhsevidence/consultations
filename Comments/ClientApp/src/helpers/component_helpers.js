// @flow

import React from "react";
import { HashLink } from "react-router-hash-link";

export const HashLinkTop = (
	label: string,
	destination: string,
	behavior: string = "instant",
	block: string = "start") => {
	return (
		<HashLink
			to={destination}
			scroll={el => el.scrollIntoView({ behavior, block })}>
			{label}
		</HashLink>
	);
};
