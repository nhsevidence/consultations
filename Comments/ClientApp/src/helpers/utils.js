import queryString from "query-string";

/**
 * Create a query string out of an object
 * @param obj the key-value pairs that you'd like to be turned into a querystring
 * @returns {string}
 */
export function objectToQueryString(obj) {
	let str = [];
	for (let p in obj)
		if (obj.hasOwnProperty(p)) {
			str.push(encodeURIComponent(p) + "=" + encodeURIComponent(obj[p]));
		}
	if (str.length) return "?" + str.join("&");
	return "";
}

export function queryStringToObject(qryString) {
	return queryString.parse(qryString);
}

/**
 * Resolve a promise on the next tick - used in testing
 * @see https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick/#process-nexttick
 * @returns {Promise<any>}
 */
export const nextTick = async () => {
	return new Promise((resolve) => {
		setTimeout(resolve, 0);
	});
};

/**
 * Format a string by replacing {0} with arguments passed in
 * @see https://stackoverflow.com/a/4673436
 * @returns {String}
 */
export function replaceFormat(stringToReplace, args) {
	if (typeof(stringToReplace) === undefined || stringToReplace === null)
		return stringToReplace;

	return stringToReplace.replace(/{(\d+)}/g, function(match, number) {
		return typeof args[number] !== undefined
			? args[number]
			: match
		;
	});
}

/**
 * @param String The string in which you want to check for the presence of "http"
 * Detects link that starts with "http"
 * @returns {Boolean}
 */
export const isHttpLink = link => link.indexOf("http") === 0;
