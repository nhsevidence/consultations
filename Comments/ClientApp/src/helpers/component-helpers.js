// @flow

import React from "react";
import { NavHashLink } from "react-router-hash-link";
import { pullFocusByQuerySelector } from "./accessibility-helpers";

type HashLinkTopType = {
	label: string,
	to: string,
	behavior: string,
	block: string
};

export function HashLinkTop(props: HashLinkTopType) {
	const { label, to, behavior, block } = props;
	return (
		<NavHashLink
			role="menuitem"
			to={to}
			scroll={el => {
				el.scrollIntoView({
					behavior,
					block,
				});
				const element = el.attributes.id.value;
				pullFocusByQuerySelector(`#${element}`);
			}}
		>
			{label}
		</NavHashLink>
	);
}

