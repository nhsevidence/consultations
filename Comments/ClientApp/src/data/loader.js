import {Endpoints, BaseUrl} from "./endpoints";
import axios from "axios";
import https from "https";
import {objectToQueryString, replaceFormat} from "./../helpers/utils";

/**
 * Load data using Axios
 * @see https://github.com/axios/axios
 * @param endpointName - can be either the shortcut of an endpoint as
 * specified in the Endpoints collection, or a URL / file path
 * @param baseUrl - the root url segment that's prepended to the query
 * @param query - the data to be serialised into query string
 * @returns {*|PromiseLike<T>|Promise<T>} - returns a promise with the data
 *
 * Example usage:
 * load("chapter", "/consultations", { consultationId: 1, documentId: 2, chapterSlug: "risk-assessment" });
 * ...would result in...
 * axios("/consultations/api/Chapter?consultationId=1&documentId=2&chapterSlug=risk-assessment").then(res => res.data);
 */

export const generateUrl = (endpointName, baseUrl = BaseUrl, urlParameters = [], query = {}) => {
	var endpoint = Endpoints[endpointName];
	if (!endpoint) return endpointName;

	return baseUrl + replaceFormat(endpoint, urlParameters) + objectToQueryString(query);
};

export const load = (endpoint, baseUrl = BaseUrl, urlParameters = [],  query = {}, method = "GET", data = {}, isJson = false) => {
	return new Promise((resolve, reject) => {
		const url = generateUrl(endpoint, baseUrl, urlParameters, query);
		const headers = isJson ? { "Content-Type" : "application/json"} : {};
		const httpsAgent = (baseUrl.indexOf("https") != -1) ? new https.Agent({ rejectUnauthorized: false }) : {};

		//baseUrl = "https://test.nice.org.uk/consultations";
		//https.globalAgent.options.rejectUnauthorized = false;
		//const httpsAgent = {}; //new https.Agent({ rejectUnauthorized: false });

		console.log(`loader.js: ${url} method: ${method}`);
		axios({url, data, method, headers, httpsAgent})
			.then(response => {
				resolve(response);
			})
			.catch(err => {
				reject(err);
			});
	});
};

export default load;
