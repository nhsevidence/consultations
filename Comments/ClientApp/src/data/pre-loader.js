import { load } from "./loader";
//import stringifyObject from "stringify-object";

// Returns data if it's available or a promise that resolves with the data
// when it's loaded async.
// This assumes the app will be rendered twice on the server: once for
// requests to fire and once when all the data has loaded
const preload = (staticContext, endpoint,  urlParameters = [], query = {}, preloadData = {}, throwOnException = true, method = "GET", headers = {}, content = null, isJson = false) => {
	let data = null;

	// Client - get data from global var
	if (typeof window !== "undefined") {
		if (!window.__PRELOADED__) return null;
		console.log("The end point is: " + endpoint);
		console.log(JSON.stringify(window.__PRELOADED__));
		data = window.__PRELOADED__[endpoint];
		delete window.__PRELOADED__[endpoint]; //this is deleted since the preloaded data should only be used on the initial page render and not persist beyond that
		console.log("Data being returned because type of window is not undefined: " + JSON.stringify(data));
		return data;
	}

	// There should always be a static context on the server but check anyway
	if (!staticContext) {
		console.warn(`Static context was null on the server when executing endpoint: ${endpoint}`);
		return data;
	}

	// Data with that key already preloaded on the server
	if (staticContext.preload.data[endpoint]) {
		console.log("Data being returned because data with that key is already preloaded on the server");
		return staticContext.preload.data[endpoint];
	}
	let cookies = "";
	if (preloadData && preloadData.cookies) {
		cookies = preloadData.cookies;
	}
	// Load fresh data on the server
	const promise = load(endpoint, staticContext.baseUrl, urlParameters, query,  method, content, isJson, cookies, headers)
		.then(response => {
			staticContext.preload.data[endpoint] = response.data;
			console.log("The endpoint from the promise: " + endpoint);
			console.log("The response data: " + JSON.stringify(response.data));
			return response.data;
		})
		.catch(err => {
			console.log("we've hit the catch");
			if (throwOnException){
				// todo: no pages loading on dev / alpha poss to do with the footer erroring...?
				 throw new Error(err);
			}
			else{
			 console.log(err); //this console.log is server-side, so not useful anywhere except locally.
			}
		});

	//staticContext.preload.data[endpoint] = data;

	// Track promises on the static context so that we can wait for all of them to resolve before
	staticContext.preload.loaders.push(promise);
	return null;
};

export default preload;
