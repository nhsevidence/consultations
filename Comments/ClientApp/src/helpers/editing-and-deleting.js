// @flow

import { load } from "../data/loader";
import { tagManager } from "./tag-manager";

export function saveCommentHandler(event: Event, comment: CommentType, self: any) {
	event.preventDefault();

	const originalId = comment.commentId;
	const isANewComment = comment.commentId < 0;
	const method = isANewComment ? "POST" : "PUT";
	const urlParameters = isANewComment ? [] : [comment.commentId];
	const endpointName = isANewComment ? "newcomment" : "editcomment";
	let error = "";

	load(endpointName, undefined, urlParameters, {}, method, comment, true)
		.then(res => {
			if (res.status === 201 || res.status === 200) {
				const index = self.state.comments
					.map(function(comment) {
						return comment.commentId;
					})
					.indexOf(comment.commentId);
				const comments = self.state.comments;
				comments[index] = res.data;
				self.setState({
					comments,
					error,
				});
				tagManager({
					event: "generic",
					category: "Consultation comments page",
					action: "Clicked",
					label: "Comment saved button",
				});
				self.updateUnsavedIds(`${originalId}c`, false);
				if (typeof self.issueA11yMessage === "function") {
					self.issueA11yMessage("Comment saved");
				}
				if (typeof self.validationHander === "function") {
					self.validationHander();
				}
			}
		})
		.catch(err => {
			console.log(err);
			if (err.response) {
				error = "save";
				self.setState({
					error,
				});
				if (typeof self.issueA11yMessage === "function") {
					self.issueA11yMessage("There was a problem saving this comment");
				}
			}
		});
}

export function deleteCommentHandler(event: Event, commentId: number, self: any) {
	event.preventDefault();
	if (commentId < 0) {
		removeCommentFromState(commentId, self);
	} else {
		load("editcomment", undefined, [commentId], {}, "DELETE")
			.then(res => {
				if (res.status === 200) {
					removeCommentFromState(commentId, self);
				}
			})
			.catch(err => {
				console.log(err);
				if (err.response) {
					const error = "delete";
					self.setState({
						error,
					});
					if (typeof self.issueA11yMessage === "function") {
						self.issueA11yMessage("There was a problem deleting this comment");
					}
				}
			});
	}
}

export function saveAnswerHandler(event: Event, answer: AnswerType, questionId: number, self: any) {
	event.preventDefault();
	const isANewAnswer = answer.answerId < 0;
	const method = isANewAnswer ? "POST" : "PUT";
	const urlParameters = isANewAnswer ? [] : [answer.answerId];
	const endpointName = isANewAnswer ? "newanswer" : "editanswer";

	load(endpointName, undefined, urlParameters, {}, method, answer, true)
		.then(res => {
			if (res.status === 201 || res.status === 200) {
				const questionIndex = self.state.questions
					.map(function(question) {
						return question.questionId;
					})
					.indexOf(answer.questionId);
				const questions = self.state.questions;

				if (questions[questionIndex].answers === null || questions[questionIndex].answers.length < 1){
					questions[questionIndex].answers = [res.data];
				} else{
					const answerIndex = questions[questionIndex].answers
						.map(function(answer) {
							return answer.answerId;
						}).indexOf(answer.answerId);

					const answers = questions[questionIndex].answers;
					answers[answerIndex] = res.data;
					questions[questionIndex].answers = answers;
				}
				self.setState({
					questions,
				});
				tagManager({
					event: "generic",
					category: "Consultation comments page",
					action: "Clicked",
					label: "Save answer button",
				});
				self.updateUnsavedIds(`${questionId}q`, false);
				if (typeof self.issueA11yMessage === "function") {
					self.issueA11yMessage("Answer saved");
				}
				if (typeof self.validationHander === "function") {
					self.validationHander();
				}
			}
		})
		.catch(err => {
			console.log(err);
			if (typeof self.issueA11yMessage === "function") {
				self.issueA11yMessage("There was a problem saving this answer");
			}
			if (err.response) alert(err.response.statusText);
		});
}

export function deleteAnswerHandler(event: Event, questionId: number, answerId: number, self: any) {
	event.preventDefault();
	if (answerId < 0) {
		removeAnswerFromState(questionId, answerId, self);
	} else {
		load("editanswer", undefined, [answerId], {}, "DELETE")
			.then(res => {
				if (res.status === 200) {
					removeAnswerFromState(questionId, answerId, self);
				}
			})
			.catch(err => {
				console.log(err);
				if (err.response) alert(err.response.statusText);
				if (typeof self.issueA11yMessage === "function") {
					self.issueA11yMessage("There was a problem deleting this answer");
				}
			});
	}
}

export function saveQuestionHandler(event: Event, question: QuestionType, self: any) {
	console.log("Called saveQuestionHandler in editing-and-deleting.js");
	console.log(event, question, self);
}

export function deleteQuestionHandler(event: Event, questionId: number, self: any) {
	console.log("Called deleteQuestionHandler in editing-and-deleting.js");
	console.log(event, questionId, self);

}

function removeCommentFromState(commentId: number, self: any) {
	self.updateUnsavedIds(`${commentId}c`, false);
	if (typeof self.issueA11yMessage === "function") {
		self.issueA11yMessage("Comment deleted");
	}
	tagManager({
		action: "Clicked",
		label: "Comment deleted button",
		event: "generic",
		category: "Consultation comments page",
	});
	let comments = self.state.comments;
	const error = "";
	comments = comments.filter(comment => comment.commentId !== commentId);
	self.setState({ comments, error });
	if ((comments.length === 0) && (typeof self.validationHander === "function")) {
		self.validationHander();
	}
}

function removeAnswerFromState(questionId: number, answerId: number, self: any) {
	self.updateUnsavedIds(`${questionId}q`, false);
	let questions = self.state.questions;
	let questionToUpdate = questions.find(question => question.questionId === questionId);
	questionToUpdate.answers = questionToUpdate.answers.filter(answer => answer.answerId !== answerId);
	self.setState({ questions });
	if (typeof self.validationHander === "function") {
		self.validationHander();
	}
	if (typeof self.issueA11yMessage === "function") {
		self.issueA11yMessage("Answer deleted");
	}
	tagManager({
		action: "Clicked",
		label: "Delete answer button",
		event: "generic",
		category: "Consultation comments page",
	});
}

function removeQuestionFromState(questionId: number, self: any) {
	console.log("Called removeQuestionFromState in editing-and-deleting.js");
	console.log(questionId, self);
}
