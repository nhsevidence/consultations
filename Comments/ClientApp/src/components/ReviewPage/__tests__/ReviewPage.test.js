/* global jest */

import React from "react";
import { mount, shallow } from "enzyme";
import { MemoryRouter } from "react-router";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import { generateUrl } from "../../../data/loader";
import { nextTick, queryStringToObject } from "../../../helpers/utils";
import ReviewPageWithRouter, {ReviewPage} from "../ReviewPage";

const mock = new MockAdapter(axios);

jest.mock("../../../context/UserContext", () => {
	return {
		UserContext: {
			Consumer: (props) => {
				return props.children({
					isAuthorised: true
				});
			}
		}
	};
});

describe("[ClientApp] ", () => {
	describe("ReviewPage Component", () => {
		const fakeProps = {
			location: {
				pathname: "",
				search: ""
			}
		};

		afterEach(() => {
			mock.reset();
		});

		it("generateDocumentList doesn't filter out documents supporting comments", async () => {
			
			const docTypesIn = [
				{title : "first doc title", sourceURI: "first source uri", supportsComments: true},
				{title : "second doc title", sourceURI: "second source uri", supportsComments: true}];

			const reviewPage = new ReviewPage(fakeProps);

			const returnValue = reviewPage.generateDocumentList(docTypesIn);

			expect(returnValue.links.length).toEqual(2);				
		});

		it("generateDocumentList filters out documents not supporting comments", async () => {
			
			const docTypesIn = [
				{title : "first doc title", sourceURI: "first source uri", supportsComments: true},
				{title : "second doc title", sourceURI: "second source uri", supportsComments: false}];

			const reviewPage = new ReviewPage(fakeProps);

			const returnValue = reviewPage.generateDocumentList(docTypesIn);

			expect(returnValue.links.length).toEqual(1);				
		});
	});
});
