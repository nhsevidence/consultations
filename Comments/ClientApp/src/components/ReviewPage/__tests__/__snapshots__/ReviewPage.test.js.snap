// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ClientApp]  ReviewPage Component should match snapshot with supplied data 1`] = `
<div
  className="container"
>
  <div
    className="grid"
  >
    <div
      data-g="12"
    >
      <aside
        className="phase-banner mt--b"
      >
        <span
          className="phase-banner__tag"
        >
          <span
            className="tag tag--impact tag--agile"
          >
            beta
          </span>
        </span>
        <span
          className="phase-banner__label"
        >
          NICE Consultations
           is in development which means that some features may not work fully.
           
          <a
            href="https://github.com/nhsevidence/consultations"
            rel="noopener noreferrer"
            target="_blank"
          >
            Report an issue
          </a>
        </span>
      </aside>
      <nav
        aria-label="Breadcrumbs"
      >
        <p
          className="visually-hidden"
        >
          You are here:
        </p>
        <ol
          className="breadcrumbs"
          itemScope={true}
          itemType="http://schema.org/BreadcrumbList"
        >
          <li
            className="breadcrumbs__crumb"
            itemProp="itemListElement"
            itemScope={true}
            itemType="http://schema.org/ListItem"
          >
            <a
              href="https://alpha.nice.org.uk/guidance/inconsultation"
            >
              All Consultations
            </a>
            <meta
              content={0}
              itemProp="position"
            />
          </li>
          <li
            className="breadcrumbs__crumb"
            itemProp="itemListElement"
            itemScope={true}
            itemType="http://schema.org/ListItem"
          >
            <a
              href="https://alpha.nice.org.uk/guidance/indevelopment/gid-dg10046/consultation/html-content"
            >
              Consultation
            </a>
            <meta
              content={1}
              itemProp="position"
            />
          </li>
          <li
            className="breadcrumbs__crumb"
            itemProp="itemListElement"
            itemScope={true}
            itemType="http://schema.org/ListItem"
          >
            <a
              href="/1/0"
              itemProp="name"
              onClick={[Function]}
            >
              Documents
            </a>
            <meta
              content={2}
              itemProp="position"
            />
          </li>
        </ol>
      </nav>
      <main
        role="main"
      >
        <div
          className="page-header"
        >
          <undefined />
          <h2
            className="mt--0"
          >
            Comments for review
          </h2>
          <div
            className="grid"
            onScroll={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              data-g="12 md:3"
            >
              <nav
                aria-label="View comments by document"
                className="stacked-nav"
              >
                <h2
                  className="stacked-nav__root"
                >
                  View comments by document
                </h2>
                <ul
                  className="stacked-nav__list"
                >
                  <li
                    className="stacked-nav__list-item"
                  >
                    <a
                      aria-current="false"
                      href="/1/review"
                      onClick={[Function]}
                    >
                      All document comments
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                  >
                    <a
                      aria-current="false"
                      href="/1/review?sourceURI=undefined"
                      onClick={[Function]}
                    >
                      Unstable angina and NSTEMI
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                  >
                    <a
                      aria-current="false"
                      href="/1/review?sourceURI=undefined"
                      onClick={[Function]}
                    >
                      Sinusitis (acute): antimicrobial prescribing
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                  >
                    <a
                      aria-current="false"
                      href="/1/review?sourceURI=undefined"
                      onClick={[Function]}
                    />
                  </li>
                  <li
                    className="stacked-nav__list-item"
                  >
                    <a
                      aria-current="false"
                      href="/1/review?sourceURI=undefined"
                      onClick={[Function]}
                    />
                  </li>
                  <li
                    className="stacked-nav__list-item"
                  >
                    <a
                      aria-current="false"
                      href="/1/review?sourceURI=undefined"
                      onClick={[Function]}
                    >
                      Unstable angina and NSTEMI
                    </a>
                  </li>
                  <li
                    className="stacked-nav__list-item"
                  >
                    <a
                      aria-current="false"
                      href="/1/review?sourceURI=undefined"
                      onClick={[Function]}
                    />
                  </li>
                  <li
                    className="stacked-nav__list-item"
                  >
                    <a
                      aria-current="false"
                      href="/1/review?sourceURI=undefined"
                      onClick={[Function]}
                    />
                  </li>
                </ul>
              </nav>
            </div>
            <div
              data-g="12 md:6"
            >
              <h3
                className="mt--0"
              >
                Comments
              </h3>
              <div>
                <p>
                  Loading...
                </p>
              </div>
            </div>
            <div
              data-g="12 md:3"
            >
              <undefined />
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
